packages/mask/background/tasks/NotCancellable/PendingTasks.ts

```diff
@@ -0,0 +1,30 @@
+import { NetworkPluginID, PersistentStorages } from '@masknet/shared-base'
+import { MessageStateType, type ReasonableMessage } from '@masknet/web3-shared-base'
+
+function checkMessages(messages: Array<ReasonableMessage<Request, Response>>) {
+    const pendingTasks = messages
+        .filter((x) => x.state === MessageStateType.NOT_DEPEND)
+        .sort((a, z) => a.createdAt.getTime() - z.createdAt.getTime())
+    const length = pendingTasks.length
+    const action = browser.action || browser.browserAction
+    action.setBadgeBackgroundColor({
+        color: '#d32f2f',
+    })
+    action.setBadgeText({
+        text: length ? length.toString() : '',
+    })
+}
+
+async function watchTasks() {
+    const { storage } = PersistentStorages.Web3.createSubScope(`${NetworkPluginID.PLUGIN_EVM}_Message`, {
+        messages: {} as Record<string, ReasonableMessage<Request, Response>>,
+    })
+    await storage.messages.initializedPromise
+    checkMessages(Object.values(storage.messages.value))
+    storage.messages.subscription.subscribe(() => {
```

Comment:
Please subscribe here: `packages/web3-providers/src/Web3/Base/state/Message.ts`.

Reply:
I think it's better to not mix them together.

Reply:
It's a part of the Message state.